### PR TITLE
several warnings

### PR DESCRIPTION
**Pull request by milbrandt**
md-lint heading style, unused import (organize imports from ide called), added some generics

Side note: `testGetYamlFile` is failing already in master on my workstation.
```
Failed tests:
  YamlSourceCodeTest.testGetYamlFile:44 expected:<---[
dict: {}
fdgsfg: fgdfg
- fdgfd:
    fgdfg: fgfgfgf
    - fgf:]
    # Comment> but was:<---[
dict: {}
fdgsfg: fgdfg
- fdgfd:
    fgdfg: fgfgfgf
]   - fgf:
    # Comment>
```

### CODE REVIEW

File: src/main/java/com/github/sbaudoin/sonar/plugins/yaml/rules/YamlRulesDefinition.java [hunk 1]
```diff
@@ -32,17 +32,17 @@
      */
     public static final String RULES_DEFINITION_FOLDER = "org/sonar/l10n/yaml/rules/yaml";
 
-
```
Comment (sbaudoin) [inline src/main/java/com/github/sbaudoin/sonar/plugins/yaml/rules/YamlRulesDefinition.java]: I usually add extra lines for readability to separate major sections in my code: statics, constructors, private methods, etc.

File: src/test/java/com/github/sbaudoin/sonar/plugins/yaml/highlighting/YamlHighlightingTest.java [hunk 2]
```diff
@@ -15,24 +15,24 @@
...
+
+import java.io.IOException;
+import java.util.Optional;
+
 import com.github.sbaudoin.sonar.plugins.yaml.Utils;
 import com.github.sbaudoin.sonar.plugins.yaml.checks.YamlSourceCode;
+
 import org.junit.Rule;
 import org.junit.Test;
-import org.sonar.api.batch.fs.InputFile;
```
Comment (sbaudoin) [inline src/test/java/com/github/sbaudoin/sonar/plugins/yaml/highlighting/YamlHighlightingTest.java]: I agree for this one but not reordering that do not bring any improvement

File: README.md [hunk 3]
```diff
@@ -14,8 +14,8 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -->
-YAML SonarQube Plugin
-=====================
```
Comment (milbrandt) [inline README.md]: I'll beg to differ a bit. Markdown headings style might not be relevant for display, but it should be consistent in the file. Only this H1 heading was with `====`, all other headings are using `#`. Therefore I would like to Keep this change.

Comment (sbaudoin) [reply README.md]: I agree but my intention is to mark document title, which is missing in MD syntax. So I'd like to keep those ===.

**Pull request merged**